modules/deriving/src/main/scala/shapeless3/deriving/kinds.scala

```diff
@@ -177,40 +177,44 @@ object K0:
   inline given mkCoproductInstances[F[_], T](using gen: CoproductGeneric[T]): CoproductInstances[F, T] =
     ErasedCoproductInstances[K0.type, F[T], LiftP[F, gen.MirroredElemTypes]](gen): CoproductInstances[F, T]
 
-object K1:
-  type Kind[C, O[_]] = C {
+abstract class K1T[TypeRestriction]:
```

Comment:
If we're doing this we should add a lower bound too. And I wouldn't call this `TypeRestriction` but `LB` and `UB` or `Lower` and `Upper`, something like that.

Reply:
I can make that change!